src/formatDate.js

```diff
@@ -7,8 +7,31 @@
  *
  * @returns {string}
  */
-function formatDate(date, fromFormat, toFormat) {
-  // write code here
+function formatDate(dateString, fromFormat, toFormat) {
+  const dateArray = dateString.split(fromFormat[3]);
+  const object = {};
+  const newDate = [];
+
+  for (let i = 0; i < 3; i++) {
+    object[fromFormat[i]] = dateArray[i];
+  }
+
+  if (object.YYYY) {
+    object.YY = object.YYYY.slice(2);
+  }
+
+  if (object.YY < 30) {
```

Comment:
This condition will throw an error if `object.YY` is not defined. You should only perform this check if you are converting from 'YY' to 'YYYY'. Also, ensure that `object.YY` is converted to a number before comparing it with 30.